lib/extensions.js

```diff
@@ -77,6 +77,28 @@ module.exports = function (realmConstructor) {
     return `${name}#${id}`;
   };
 
+  // Overriding Object.is()
+  Object.is = function (x, y) {
+    if (x instanceof realmConstructor.Object && y instanceof realmConstructor.Object) {
+      return x._objectId() === y._objectId();
+    }
+
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+    // SameValue algorithm
+    if (x === y) {
+      // return true if x and y are not 0, OR
+      // if x and y are both 0 of the same sign.
+      // This checks for cases 1 and 2 above.
+      return x !== 0 || 1 / x === 1 / y;
+    } else {
+      // return true if both x AND y evaluate to NaN.
+      // The only possibility for a variable to not be strictly equal to itself
+      // is when that variable evaluates to NaN (example: Number.NaN, 0/0, NaN).
+      // This checks for case 3.
+      return x !== x && y !== y;
+    }
```

Comment:
I would definitely just store the `const originalObjectIs = Object.is.bind(Object);` and defer to that instead of writing our own fallback.